eng/native/init-os-and-arch.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+# Use uname to determine what the OS is.
+OSName=$(uname -s)
+case "$OSName" in
+    Darwin)
+        os=OSX
+        ;;
+
+    FreeBSD)
+        os=FreeBSD
+        ;;
+
+    Linux)
+        os=Linux
+        ;;
+
+    NetBSD)
+        os=NetBSD
+        ;;
+
+    OpenBSD)
+        os=OpenBSD
+        ;;
+
+    SunOS)
+        os=SunOS
+        ;;
+
+    *)
+        echo "Unsupported OS $OSName detected, configuring as if for Linux"
+        os=Linux
+        ;;
+esac
+
+# Use uname to determine what the CPU is.
+CPUName=$(uname -p)
+
+# Some Linux platforms report unknown for platform, but the arch for machine.
+if [[ "$CPUName" == "unknown" ]]; then
+    CPUName=$(uname -m)
+fi
+
+case "$CPUName" in
+    aarch64)
+        arch=arm64
+        ;;
+
+    amd64)
+        arch=x64
+        ;;
+
+    armv7l)
+        if (NAME=""; . /etc/os-release; test "$NAME" = "Tizen"); then
+            __BuildArch=armel
+            __HostArch=armel
+        else
+            __BuildArch=arm
+            __HostArch=arm
+        fi
+        ;;
+
+    i686)
```

Comment:
Might want to case on `i[3-6]86` here instead, to be thorough.  (Even though machines older than i686 are *really* unlikely these days.)